jsonnet/kube-thanos/kube-thanos-store.libsonnet

```
@@ -1,111 +1,162 @@
-          container.mixin.readinessProbe.httpGet.withPath('/-/ready');
+    sts.new(ts.config.name, ts.config.replicas, c, [], ts.config.commonLabels) +
+    sts.mixin.metadata.withNamespace(ts.config.namespace) +
+    sts.mixin.metadata.withLabels(ts.config.commonLabels) +
+    sts.mixin.spec.withServiceName(ts.service.metadata.name) +
+    sts.mixin.spec.template.spec.withTerminationGracePeriodSeconds(120) +
+    sts.mixin.spec.template.spec.withVolumes([
+      volume.fromEmptyDir('data'),
+    ]) +
+    sts.mixin.spec.selector.withMatchLabels(ts.config.podLabelSelector) +
+    {
+      spec+: {
+        volumeClaimTemplates: null,
+      },
+    },
+
+  withServiceMonitor:: {
+    local ts = self,
+    serviceMonitor: {
+      apiVersion: 'monitoring.coreos.com/v1',
+      kind: 'ServiceMonitor',
+      metadata+: {
+        name: ts.config.name,
+        namespace: ts.config.namespace,
+        labels: ts.config.commonLabels,
+      },
+      spec: {
+        selector: {
+          matchLabels: ts.config.podLabelSelector,
+        },
+        endpoints: [
+          { port: 'http' },
+        ],
+      },
+    },
+  },
 
-        sts.new(ts.name, ts.replicas, c, [], $.thanos.store.statefulSet.metadata.labels) +
-        sts.mixin.metadata.withNamespace(ts.namespace) +
-        sts.mixin.metadata.withLabels({ 'app.kubernetes.io/name': ts.name }) +
-        sts.mixin.spec.withServiceName($.thanos.store.service.metadata.name) +
-        sts.mixin.spec.selector.withMatchLabels($.thanos.store.statefulSet.metadata.labels) +
-        sts.mixin.spec.template.spec.withTerminationGracePeriodSeconds(120) +
-        sts.mixin.spec.template.spec.withVolumes([
-          volume.fromEmptyDir(ts.name + '-data'),
-        ]) +
-        {
+  withVolumeClaimTemplate:: {
+    local ts = self,
+    config+:: {
+      volumeClaimTemplate: error 'must provide volumeClaimTemplate',
+    },
+    statefulSet+: {
+      spec+: {
+        template+: {
           spec+: {
-            volumeClaimTemplates:: null,
+            volumes: null,
           },
         },
+        volumeClaimTemplates: [ts.config.volumeClaimTemplate {
+          metadata+: {
+            name: 'data',
+          },
+        }],
+      },
     },
+  },
 
-    querier+: {
-      deployment+: {
-        spec+: {
-          template+: {
-            spec+: {
-              containers: [
-                super.containers[0]
-                { args+: [
-                  '--store=dnssrv+_grpc._tcp.%s.%s.svc.cluster.local' % [
-                    $.thanos.store.service.metadata.name,
-                    $.thanos.store.service.metadata.namespace,
-                  ],
-                ] },
-              ],
-            },
+  withResources:: {
+    local ts = self,
+    config+:: {
+      resources: error 'must provide resources',
+    },
+
+    statefulSet+: {
+      spec+: {
+        template+: {
+          spec+: {
+            containers: [
+              if c.name == 'thanos-store' then c {
```

This needs to use the templated name :)
```suggestion
              if c.name == ts.config.name then c {
```

---

whops, never intended the container name to be configurable

---

very good catch!

---

fixed